dspace-api/src/test/java/org/dspace/util/ProxyUtilsTest.java

```diff
@@ -0,0 +1,136 @@
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.withSettings;
+
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+
+import org.apache.http.HttpHost;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.dspace.AbstractUnitTest;
+import org.dspace.services.ConfigurationService;
+import org.dspace.utils.DSpace;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * Test utilities for setting outgoing HTTP proxies.
+ *
+ * @author mwood
+ */
+public class ProxyUtilsTest
+        extends AbstractUnitTest {
+    private static final String PROXY_HOST = "proxy.example.com";
+    private static final String PROXY_IP_ADDRESS = "192.168.1.1";
+    private static final String PROXY_PORT = "8888";
+    private static final int PROXY_PORT_INT = Integer.parseInt(PROXY_PORT);
+
+    private static ConfigurationService cfg;
+    private static String old_host;
+    private static String old_port;
+
+    @BeforeClass
+    public static void setup() {
+        cfg = new DSpace().getConfigurationService();
+        old_host = cfg.getProperty(ProxyUtils.C_HTTP_PROXY_HOST);
+        old_port = cfg.getProperty(ProxyUtils.C_HTTP_PROXY_PORT);
+    }
+
+    @AfterClass
+    public static void shutdown() {
+        cfg.setProperty(ProxyUtils.C_HTTP_PROXY_HOST, old_host);
+        cfg.setProperty(ProxyUtils.C_HTTP_PROXY_PORT, old_port);
+    }
+
+    /**
+     * Test of addProxy method when no proxy is set.
+     */
+    @Test
+    public void testAddProxyNone() {
+        HttpClientBuilder builder = HttpClientBuilder.create();
+        HttpClientBuilder builderSpy = spy(builder);
+
+        // Test no proxy
+        cfg.setProperty(ProxyUtils.C_HTTP_PROXY_HOST, null);
+        cfg.setProperty(ProxyUtils.C_HTTP_PROXY_PORT, null);
+        builderSpy = ProxyUtils.addProxy(builderSpy);
+        verify(builderSpy,never()).setProxy(any(HttpHost.class));
+    }
+
+    /**
+     * Test of addProxy method when only host is set.
+     */
+    @Test
+    public void testAddProxyNoPort() {
+        HttpClientBuilder builder = HttpClientBuilder.create();
+        HttpClientBuilder builderSpy = spy(builder);
+        HttpHost hostSpy = mock(HttpHost.class,
+                withSettings().useConstructor("bogus." + PROXY_HOST, Integer.valueOf(-1)));
+
+        cfg.setProperty(ProxyUtils.C_HTTP_PROXY_HOST, PROXY_HOST);
+        cfg.setProperty(ProxyUtils.C_HTTP_PROXY_PORT, null);
+        builderSpy = ProxyUtils.addProxy(builderSpy);
+        verify(builderSpy,never()).setProxy(any(HttpHost.class));
+        // FIXME this fails with null host name.  How to get at the HttpHost?
+        // FIXME assertEquals("Wrong proxy host", PROXY_HOST, hostSpy.getHostName());
+        // FIXME assertEquals("Wrong proxy port", PROXY_PORT, hostSpy.getPort()); // FIXME nope?
```

Comment:
I'd recommend either removing these FIXME comments, or adding more details about *how* to fix these issues in the future.  Currently, they just sound like a conversation with yourself...which is fine for development, but not something we want to add into core code.